components/DatePicker/DatePickerRange.js

```diff
@@ -1,24 +1,92 @@
 import { useState } from "react";
 import DatePicker from "react-datepicker";
+import "react-datepicker/dist/react-datepicker.css";
+import { eachDayOfInterval, isValid } from "date-fns";
 
-export default function DatePickerRange({ startDate, endDate, onChange }) {
+export default function DatePickerRange({
+  startDate,
+  endDate,
+  onChange,
+  excludeDates,
+}) {
   const [dateRange, setDateRange] = useState([startDate, endDate]);
+  const [isOpen, setIsOpen] = useState(false);
+  const today = new Date();
+  const nextMonthLastDate = new Date(
+    today.getFullYear(),
+    today.getMonth() + 2,
+    0
+  );
+  const [selectedDateRanges, setSelectedDateRanges] = useState([]);
+
+  function getDatesBetween(start, end) {
+    const dates = [];
+    let currentDate = new Date(start);
+
+    while (currentDate <= end) {
+      dates.push(new Date(currentDate));
+      currentDate.setDate(currentDate.getDate() + 1);
+    }
+
+    return dates;
+  }
 
   //updating the state of this component
   function handleDateChange(update) {
     setDateRange(update);
-    onChange(update);
+    setSelectedDateRanges((prevRanges) => [...prevRanges, update]);
+
+    const allDates = getDatesBetween(update[0], update[1]);
+
+    onChange(allDates);
+    console.log("Date range changed:", allDates);
+  }
+
+  function handleClick(e) {
+    e.preventDefault();
+    setIsOpen(!isOpen);
+  }
+  function handleReset() {
+    setDateRange([]);
   }
 
   return (
-    <DatePicker
-      selectsRange={true}
-      dateFormat="dd/MM/yyyy"
-      startDate={dateRange[0]}
-      endDate={dateRange[1]}
-      onChange={handleDateChange}
-      isClearable={true}
-      placeholderText="Choose your dates"
-    />
+    <>
+      <button
+        type="button"
+        className="example-custom-input"
```

Comment:
Since we don't style elements via `className` you'll need to remove it.

Reply:
Here was a suggestion from the React DatePicker documentation! I will try to modify it.

Reply:
Ah, I see.